libp2p/builders.nim

```diff
@@ -166,7 +171,9 @@ proc build*(b: SwitchBuilder): Switch
 
   let
     identify = Identify.new(peerInfo)
-    connManager = ConnManager.new(b.maxConnsPerPeer, b.maxConnections, b.maxIn, b.maxOut)
+    maxIn = if b.maxIn >= 0: b.maxIn else: b.maxConnections
+    maxOut = if b.maxOut >= 0: b.maxOut else: b.maxConnections
+    connManager = ConnManager.new(b.maxConnsPerPeer, b.maxConnections, maxIn, maxOut, b.forceDial)
```

Comment:
why state and not an argument to the `dial` function? I can imagine applications wanting to control this on a dial-by-dial basis